mllib/src/main/scala/org/apache/spark/mllib/tree/DecisionTree.scala

```diff
@@ -58,8 +67,8 @@ class DecisionTree @Since("1.0.0") (private val strategy: Strategy)
    */
   @Since("1.2.0")
   def run(input: RDD[LabeledPoint]): DecisionTreeModel = {
-    // Note: random seed will not be used since numTrees = 1.
-    val rf = new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all", seed = 0)
+    val rf = new RandomForest(strategy, numTrees = 1, featureSubsetStrategy = "all",
```

Comment:
Across the tree/ensemble libraries in Spark, unit tests generally test that ML and MLlib results are equal. For decision trees, ML uses a random seed and MLlib basically ignores it. By some stroke of luck, this doesn't cause a problem. This is because `findSplits` only subsamples continuous features for split calculations on large datasets using a random seed and the unit tests do not deal with large enough data sets. However, GBTs use subsampling of the input on single decision trees (this is not allowed or tested in DecisionTree*) which also uses the random seed. The unit tests fail because MLlib doesn't use the same random seed as ML. Instead of re-writing the tests, it was a fairly simple fix to add a random seed to GBT in MLlib, which required adding it to MLlib decision trees as well.